scripts/us_census/acs5yr/subject_tables/common/generate_col_map.py

```diff
@@ -0,0 +1,118 @@
+  def __init__(self, specDict=None, columnList=None):
+    """module init """
+    self.features = specDict
+    self.column_list = columnList
+    self.column_map = {}
+
+  def _generate_stat_vars_from_spec(self):
+    """generates stat_var nodes for each column in column list and returns a dictionary of columns with their respective stat_var nodes"""
+    # for each column generate the definition of their respective statistical variable node
+    for col in self.column_list:
+      if 'ignoreColumns' in self.features and col not in self.features['ignoreColumns']:
+          self.column_map[col] = self._column_to_statvar(col)
+    #TODO: Enable validating the generate columnMap: _valid_col_map():
+    return self.column_map
+
+  def _column_to_statvar(self, column):
+    """generates a dictionary statistical variable with all properties specified in the JSON spec for a single column"""
+    measurement_assigned = False
+    stat_var = {}
+    # part_list contains a list of tokens of the column name after splitting the
+    # column name based on the '!!' delimiter
+    part_list = column.split('!!')
+    # part_list[-1], the last element of part_list describes the property value.
+    # This element points to the 'v' in 'pv' -> and we map it to the
+    # corresponding property 'p'
+    val_prop_col = part_list[-1]
+
+    for part in part_list:
+      # set the base for special cases like median, etc.
+      if not measurement_assigned and 'measurement' in self.features.keys():
+        if part in self.features['measurement'].keys():
+          stat_var.update(self.features['measurement'][part])
+          measurement_assigned = True
+
+    # set the default statVar definition
+    if not measurement_assigned and 'measurement' in self.features.keys() and '_DEFAULT' in self.features['measurement'].keys():
+      stat_var.update(self.features['measurement']['_DEFAULT'])
+
+    # set the populationType attribute
+    stat_var['populationType'] = self._get_population_type(part_list)
+
+    # associate pvs to stat_var
+    if val_prop_col not in self.features['measurement'].keys():
+      if 'pvs' in self.features.keys():
+        #p = property, k = dictionary of substrings in column that points to a propertyValue
+        for p, k in self.features['pvs'].items():
+          for c, v in k.items():
+            ## check if the val_prop_col and the current key of the dict match 
+            if val_prop_col == c:
+              stat_var[p] = v
+              # add inferred properties from property 'p' to the stat_var
+              if 'inferredSpec' in self.features.keys() and p in self.features['inferredSpec'].keys():
+                stat_var.update(self.features['inferredSpec'][p])
+
+            ## if the above fails, check if the current key of dict matched with a token's substring
+            ## Eg: In table S2702, 'Uninsured' is the substring in the column that maps to the value 'NoHealthInsurance'.
```

Comment:
why don't we list "Total Uninsured" as the key so it will be covered by the condition above?